tests/template_tests/test_loaders.py

```diff
@@ -50,11 +50,30 @@ def test_get_template(self):
         self.assertEqual(template.origin.loader, self.engine.template_loaders[0].loaders[0])
 
     def test_get_template_missing(self):
+        self.engine.debug = True
         with self.assertRaises(TemplateDoesNotExist):
             self.engine.get_template('doesnotexist.html')
         e = self.engine.template_loaders[0].get_template_cache['doesnotexist.html']
         self.assertEqual(e.args[0], 'doesnotexist.html')
 
+        self.engine.debug = False
+        with self.assertRaises(TemplateDoesNotExist):
+            self.engine.get_template('alsodoesnotexist.html')
+        e = self.engine.template_loaders[0].get_template_cache['alsodoesnotexist.html']
+        self.assertEqual(e, TemplateDoesNotExist)
+
+    def test_cached_exception_no_traceback(self):
+        self.engine.debug = True
+        with self.assertRaises(TemplateDoesNotExist):
+            self.engine.get_template('cachestemplateresponse.html')
+        e = self.engine.template_loaders[0].get_template_cache['cachestemplateresponse.html']
+
+        if six.PY3:
```

Comment:
alternatively could use `if six.PY2: return` so we can simply remove the condition instead of also deindenting some lines when dropping Python 2.